pkg/openstack/cinder.go

```diff
@@ -17,17 +17,21 @@ import (
 
 // ReconcileCinder -
 func ReconcileCinder(ctx context.Context, instance *corev1beta1.OpenStackControlPlane, helper *helper.Helper) (ctrl.Result, error) {
-	if !instance.Spec.Cinder.Enabled {
-		return ctrl.Result{}, nil
-	}
-
 	cinder := &cinderv1.Cinder{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      "cinder",
 			Namespace: instance.Namespace,
 		},
 	}
 
+	if !instance.Spec.Cinder.Enabled {
+		if res, err := CheckDeleteDisabled(ctx, helper, cinder); err != nil {
+			return res, err
+		}
+		instance.Status.Conditions.Remove(corev1beta1.OpenStackControlPlaneCinderReadyCondition)
```

Comment:
+1

I think this allow us to make IsReady() simpler see my comment there.

Reply:
Could we set this back to `Unknown` as an alternative to removing it outright?

Reply:
I think "removing the ready status" when the service does not exist/disabled, rather than setting it to 'Unknown' makes more sense. Also, we can look at implementing it in many different ways, but I would probably get something in that's working and then followup with whatever optimization we want to do.

Reply:
Yeah I agree with @rabi. A service condition with Unknown here means the service needs to be deployed but the deploy hasn't been started yet. When the service is disabled we should not really add the service condition to the list as it is misleading.